Metric/MetricHandler.php

```diff
@@ -98,7 +98,10 @@ public function setContainer(ContainerInterface $container): void
 
     public function setRequestStack(RequestStack $requestStack): void
     {
-        $this->requestStack = $requestStack;
+        //We clone the request stack when it is instantiated.
+        //We will need to use the request stack during the kernel terminate phase and the request stack is now
+        //empty before this phase. So if we want to keep the original object state, we need to break its reference.
+        $this->requestStack = clone $requestStack;
```

Comment:
Why don't you retrieve immediately the master request? As far as I can see, it's the only purpose of the request stack injection.